network-config-analyzer/K8sService.py

```diff
@@ -0,0 +1,79 @@
+#
+# Copyright 2020- IBM Inc. All rights reserved
+# SPDX-License-Identifier: Apache2.0
+#
+
+from enum import Enum
+import Peer
+
+class K8sService:
+    """
+    Represents a K8s Service, storing its parameters
+    """
+
+    class ServiceType(Enum):
+        ClusterIP = 0 # default
+        NodePort = 1
+        LoadBalancer = 2
+        ExternalName = 3
+
+    class ServicePort:
+        """
+        Represents a K8s Service port
+        """
+        def __init__(self, port, targetPort, protocol, name=''):
+            self.port = port
+            # a target port may be given either as a number or as a string (named port)
+            if isinstance(targetPort, int):
+                self.targetPortNumber = targetPort
+            else:
+                self.targetPortName = targetPort
+            self.protocol = protocol
+            self.name = name
+
+    def __init__(self, name):
+        self.name = name
+        self.labels = {}  # Storing the namespace labels in a dict as key-value pairs
+        self.type = self.ServiceType.ClusterIP
+        self.selector = {}
+        self.ports = {}  # a map from service port name to ServicePort object
+        self.target_pods = Peer.PeerSet()
+
+    def __eq__(self, other):
+        if isinstance(other, K8sService):
+            return self.name == other.name
+        return NotImplemented
+
+    def __hash__(self):
+        return hash(self.name)
+
+    def __str__(self):
+        return self.name
+
+    def set_type(self, type):
```

Comment:
`type` is a Python built-in function - better rename

Reply:
Done